src/main.nim

```diff
@@ -5870,13 +5871,28 @@ proc handleGlobalKeyEvents(a) =
       elif ke.isShortcutDown(scJumpToLinkedCell):
         let otherLocs = map.getLinkedLocations(cur)
         if otherLocs.len == 1:
-          moveCursorTo(otherLocs.first.get, a)
+          let otherLoc = otherLocs.first.get
+          if map.getLinkedLocations(otherLoc).len > 1:
+            ui.lastJumpToSrcLocation = cur
+          moveCursorTo(otherLoc, a)
         elif otherLocs.len > 1:
-          ui.jumpToDestLocation = cur
           ui.jumpToSrcLocation = otherLocs.toSeq
           sort(ui.jumpToSrcLocation)
-          ui.jumpToSrcLocationIdx = 0
-          moveCursorTo(ui.jumpToSrcLocation[ui.jumpToSrcLocationIdx], a)
+          if ui.jumpToDestLocation == cur:
+            # Last time we jumped to multiple sources we used this destination,
+            # so continue selecting sources from the source we left at.
+            let oldIdx = ui.jumpToSrcLocation.find(ui.lastJumpToSrcLocation)
+            if oldIdx == -1:
+              # The source we left at last time no longer exists (e.g. the user
+              # deleted it), so reset from beginning.
+              ui.jumpToSrcLocationIdx = 0
```

Comment:
Nice that you thought of this case 👍🏻 It can easily happen with undo/redo as well.